frontend/www/js/omegaup/components/common/GridPaginator.vue

```diff
@@ -0,0 +1,134 @@
+    </h5>
+    <div class="card-body text-center" v-if="sortOptions.length > 0">
+      <div class="form-check form-check-inline">
+        <label
+          v-bind:key="index"
+          v-for="(sortOption, index) in sortOptions"
+          class="form-check-label mr-4"
+        >
+          <input
+            class="form-check-input m-0"
+            name="sort-selector"
+            type="radio"
+            v-bind:value="sortOption.value"
+            v-model="currentSortOption"
+          />
+          {{ sortOption.title }}
+        </label>
+      </div>
+    </div>
+    <table class="table table-striped mb-0" v-if="items.length > 0">
+      <slot name="table-header"></slot>
+      <tbody>
+        <tr v-bind:key="index" v-for="(group, index) in paginatedItems">
+          <th scope="row" v-if="showPageOffset" class="text-center">
+            {{ currentPageNumber * rowsPerPage + (index + 1) }}
+          </th>
+          <td v-bind:key="itemIndex" v-for="(item, itemIndex) in group">
+            <slot name="item-data" v-bind:item="item">
+              <a v-bind:href="item.getUrl()">
+                {{ item.toString() }}
+              </a>
+            </slot>
+          </td>
+          <td class="text-right" v-if="!group[0].getBadge().isEmpty()">
+            <strong>{{ group[0].getBadge().get() }}</strong>
+          </td>
+        </tr>
+      </tbody>
+    </table>
+    <div class="card-footer text-center" v-if="items.length > 0">
+      <div class="btn-group" role="group">
+        <button
+          class="btn btn-primary"
+          type="button"
+          v-bind:disabled="
+            this.totalPagesCount === 1 || this.currentPageNumber === 0
+          "
+          v-on:click="previousPage"
+        >
+          {{ T.wordsPrevious }}
+        </button>
+        <button
+          class="btn btn-primary"
+          type="button"
+          v-bind:disabled="this.totalPagesCount === 1 || this.currentPageNumber &gt;= this.totalPagesCount - 1"
+          v-on:click="nextPage"
+        >
+          {{ T.wordsNext }}
+        </button>
+      </div>
+    </div>
+  </div>
+</template>
+
+<script lang="ts">
+import { Vue, Component, Prop, Watch } from 'vue-property-decorator';
+import T from '../../lang';
+import { LinkableResource } from '../../types';
+
+interface SortOption {
+  value: string;
+  title: string;
+}
+
+/**
+  Creates a two-dimensional paginated table, with the number of columns passed
+  as a prop and the number of rows being calculated taking into account the number
+  of items per page, total items and the number of columns.
+ */
+@Component
+export default class GridPaginator extends Vue {
+  @Prop() items!: LinkableResource[];
+  @Prop() itemsPerPage!: number;
+  @Prop({ default: 3 }) columns!: number;
+  @Prop() title!: string;
+  @Prop({ default: false }) showPageOffset!: boolean;
+  @Prop({ default: () => [] }) sortOptions!: SortOption[];
```

Comment:
hmm esto está raro: el tipo de la variable es `SortOption[]`, pero el tipo del default es "función que regresa `SortOption[]`". qué pasa si se cambia a:

```suggestion
  @Prop({ default: [] }) sortOptions!: SortOption[];
```

? y ahora me pregunto por qué TypeScript no se está quejando :/

Reply:
Es por esto 
![image](https://user-images.githubusercontent.com/20785082/81831581-e826de00-9513-11ea-8e04-2d9bbb9dee62.png)

Reply:
ya veo, gracias!